searx/engines/hackernews.py

```diff
@@ -0,0 +1,98 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""
+Hackernews
+"""
+
+from datetime import datetime, timedelta
+from urllib.parse import urlencode
+
+# Engine metadata
+about = {
+    "website": "https://news.ycombinator.com/",
+    "wikidata_id": "Q686797",
+    "official_api_documentation": "https://hn.algolia.com/api",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+# Engine configuration
+paging = True
+time_range_support = True
+categories = ["it"]
+results_per_page = 20
+
+# Search URL
+base_url = "https://hn.algolia.com/api/v1/search"
+
+
+def request(query, params):
+    if query.lower() in ("front_page", "fp"):
+        query_params = {"tags": "front_page"}
+    else:
+        query_params = {
+            "query": query,
+            "page": params["pageno"],
+            "hitsPerPage": results_per_page,
+        }
+
+        if 'time_range' in params:
+            timestamp = calculate_timestamp(params['time_range'])
+            if timestamp is not None:
+                query_params["numericFilters"] = f"created_at_i>{timestamp}"
+
+    params["url"] = f"{base_url}?{urlencode(query_params)}"
+    return params
+
+
+def calculate_timestamp(time_range):
+    time_ranges = {'year': 365, 'month': 30, 'week': 7, 'day': 1}
+    if time_range in time_ranges:
+        days = time_ranges[time_range]
+        return int((datetime.utcnow() - timedelta(days=days)).timestamp())
+    return None
+
+
+def time_ago(created_at_i):
+    units = [("year", 31536000), ("month", 2592000), ("day", 86400)]
+    time_difference = int(datetime.utcnow().timestamp()) - created_at_i
+
+    for unit, seconds_in_unit in units:
+        if time_difference >= seconds_in_unit:
+            count = int(time_difference // seconds_in_unit)
+            return f"{count} {unit}{'s' if count > 1 else ''} ago"
+
+    return "past day"
+
+
+def response(resp):
+    results = []
+    data = resp.json()
+
+    for hit in data["hits"]:
+        object_id, title, url, points, author, created_at_i, num_comments = (
+            hit["objectID"],
+            hit["title"],
+            hit["url"],
+            hit["points"],
+            hit["author"],
+            hit["created_at_i"],
+            hit["num_comments"],
+        )
+
+        url = f"https://news.ycombinator.com/item?id={object_id}" if url is None or not url else url
+        title = f"post by {author}" if title is None or not title else title
+        num_comments = 0 if num_comments is None or not num_comments else num_comments
+        points = 0 if points is None or not points else points
+
+        content = (
+            f"{points} point{'s' if points != 1 else ''} | "
+            f"{author} | "
+            f"{time_ago(created_at_i)} | "
+            f"{num_comments} comment{'s' if num_comments != 1 else ''} | "
+            f"https://news.ycombinator.com/item?id={object_id}"
```

Comment:
This url shouldn't be part of the content

Reply:
i actually thought to use this for the url instead, because i like to read the discussion on hackernews, there is no way to find it without. @return42 what do you think?

Reply:
Since this engine is for hackernews, users would expect links to hackernews .. or I'm wrong?